src/imitation/algorithms/adversarial.py

```diff
@@ -352,10 +395,28 @@ def init_trainer(env_name: str,
     init_rl_kwargs: Keyword arguments passed to `init_rl`,
         used to initialize the RL algorithm.
   """
+  # come up with an informative, unique name for this expt
+  log_prefix = log_dir if log_dir is not None else 'output'
+  date_str = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M')
+  method_name = 'gail' if use_gail else 'airl'
+  log_base = os.path.join(
+    log_prefix, f'{method_name}_{env_name}_{date_str}')
+  os.makedirs(os.path.dirname(log_base), exist_ok=True)
+  i = 0
+  while True:
+    # find a short, unique suffix
+    full_log_dir = log_base + f"_run{i:02d}/"
+    try:
+      os.makedirs(full_log_dir, exist_ok=False)
+      break
+    except FileExistsError:
+      pass
+    i += 1
+  log_dir = full_log_dir
```

Comment:
Yep, that's fair. Maybe `log_dir` (or `full_log_dir`, or whatever) should be a mandatory argument instead of a kwarg? If we can't have a sensible automatic default that works for all use cases then we should not have a default at all.

Reply:
Fixed by forcing caller to pass in `log_dir` in full.